x-pack/solutions/security/plugins/security_solution_serverless/public/navigation/side_navigation.tsx

```diff
@@ -0,0 +1,909 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import * as Rx from 'rxjs';
+
+import { ExternalPageName, NavigationProvider } from '@kbn/security-solution-navigation';
+import type { NavigationTreeDefinition } from '@kbn/core-chrome-browser';
+import { i18n } from '@kbn/i18n';
+import { EuiCallOut, EuiFlexGroup, EuiFlexItem, EuiSpacer } from '@elastic/eui';
+import { LinkButton } from '@kbn/security-solution-navigation/links';
+import { type Services } from '../common/services';
+
+const SOLUTION_NAME = i18n.translate(
+  'xpack.securitySolutionServerless.navLinks.projectType.title',
+  { defaultMessage: 'Security' }
+);
+
+const PROJECT_SETTINGS_TITLE = i18n.translate(
+  'xpack.securitySolutionServerless.navLinks.projectSettings.title',
+  { defaultMessage: 'Project Settings' }
+);
+
+const createNavigationTree$ = (services: Services): Rx.Observable<NavigationTreeDefinition> => {
+  return Rx.of({
+    body: [
+      {
+        type: 'navGroup',
+        id: 'security_solution_nav',
+        title: SOLUTION_NAME,
+        icon: 'logoSecurity',
+        breadcrumbStatus: 'hidden',
+        defaultIsCollapsed: false,
+        children: [
+          {
+            breadcrumbStatus: 'hidden',
+            children: [
+              {
+                id: 'discover:',
+                link: 'discover',
+              },
+              {
+                id: 'dashboards',
+                link: 'securitySolutionUI:dashboards',
```

Comment:
Would it make sense to use the `SecurityPageName` enum to generate the security `link` values? Something like:

```ts
const securityLink = (pageName: SecurityPageName): AppDeepLinkId => {
  return `securitySolutionUI:${pageName}`;
};
```

So here we would be able to do:
```suggestion
                id: SecurityPageName.dashboards,
                link: securityLink(SecurityPageName.dashboards),
```

This will allow us to automatically apply any change in the enum values to the nav tree, and this will also make it easier to find the link names via intellisense in the IDE. 

Only for internal Security links.
 
wdyt?

Reply:
Yes, this makes sense. There are 62 of these links in the declaration, so having a helper will definitely make this more maintainable.

This gives the declaration some slight dynamism, so I will also add a unit test for each declaration file.